x/autopilot/types/autopilot.go

```diff
@@ -8,17 +8,18 @@ import (
 )
 
 // RawPacketMetadata defines the raw JSON memo that's used in an autopilot transfer
-// The PFM forward key is also used here to validate that the packet is not trying
-// to use autopilot and PFM at the same time
-// As a result, only the forward key is needed, cause the actual parsing of the PFM
-// packet will occur in the PFM module
+// The PFM forward key and wasm keys are also used here to validate that the packet
+// was not trying to use more than one of autopilot PFM, and wasm at the same time
+// As a result, only the key is needed, cause the actual parsing of the PFM/wasm
+// packet will occur in the respective module
 type RawPacketMetadata struct {
 	Autopilot *struct {
 		Receiver string                  `json:"receiver"`
 		Stakeibc *StakeibcPacketMetadata `json:"stakeibc,omitempty"`
 		Claim    *ClaimPacketMetadata    `json:"claim,omitempty"`
 	} `json:"autopilot"`
 	Forward *interface{} `json:"forward"`
+	Wasm    *interface{} `json:"wasm"`
```

Comment:
The main thing I was confused about before was that normal packets could be picked up and interpreted by a wasm contract but the diagram you made in excalidraw of how all the middlewares interact helped a lot.  I was confused because I thought before that only tx specifically aimed at cosmwasm modules would be seen by wasm but now I see how it all fits together and they use different parts of the data so that it is possible for these cosm middlewares and another middleware like autopilot can act on the same packets even using data from the same memo.  Unit tests also made it clearer